src/tcgwars/logic/impl/gen7/SunMoonPromos.groovy

```diff
@@ -2908,12 +3201,122 @@ public enum SunMoonPromos implements LogicCardInfo {
+}
+            }
+          }
+          move "Leaf Blade", {
+            text "80+ damage. Flip a coin. If heads, this attack does 40 more damage."
+            energyCost G, C, C
+            onAttack {
+              damage 80
+              flip { damage 40 }
+            }
+          }
+        };
+      case GLACEON_SM238:
+        return evolution (this, from:"Eevee", hp:HP110, type:W, retreatCost:2) {
+          weakness M
+          move "Snow Cloak", {
+            text "30 damage. Flip a coin. If heads, prevent all effects of attacks, including damage, done to this Pokémon during your opponent's next turn."
+            energyCost W
+            attackRequirement {}
+            onAttack {
+              damage 30
+              flip { peventAllEffectsNextTurn }
+            }
+          }
+          move "Hypnotic Blizzard", {
+            text "90 damage. Your opponent's active Pokémon is now Asleep. This attack does 20 damage to each of your opponent's Benched Pokémon. (Don't apply Weakness and Resistance for Benched Pokémon.)"
+            energyCost W, W, C
+            attackRequirement {}
+            onAttack {
+              damage 90
+              apply ASLEEP
+              opp.bench.each {
+                damage 20, it
+              }
+            }
+          }
+        };
+      case CARRACOSTA_GX_SM239:
+        return evolution (this, from:"Tirtouga", hp:HP250, type:F, retreatCost:4) {
+          weakness G
+          bwAbility "High Density Armor", {
+            text "If this Pokémon has full HP, it takes 90 less damage from your opponent’s attacks (after applying Weakness and Resistance)."
+            delayedA {
+              before APPLY_ATTACK_DAMAGES, {
+                bg.dm().self {
+                  if (self.damage == 0) {
+                    bc "High Density Armor -90"
+                    it.dmg -= hp(90)
+                  }
+                }
+              }
+            }
+          }
+          move "Ground Crush", {
+            text "160 damage. The Defending Pokémon can’t retreat during your opponent’s next turn."
+            energyCost F, C, C, C
+            onAttack {
+              damage 160
+              cantRetreat, defending
+            }
+          }
+          move "Stone Age GX", {
+            text "Put any number of Pokémon that evolve from Unidentified Fossil from your discard pile onto your Bench. (You can’t use more than 1 GX attack in a game.)"
+            energyCost C
+            attackRequirement {
+              gxCheck()
+              assert my.discard.findAll { it.cardTypes.is(EVOLUTION) && it.predecessor == "Unidentified Fossil" } : "No Pokémon that evolve from Unidentified Fossil in your discard pile."
+            }
+            onAttack {
+              gxPerform()
+              def maxSpace = my.bench.freeBenchCount
+                discard.search(max: maxSpace, "Bench up to $maxSpace card(s) from the discard that evolve from Unidentified Fossil.", {
```

Comment:
I think this is missing a closing bracket

Reply:
Fixed this